suite-native/formatters/src/components/EthereumTokenToFiatAmountFormatter.tsx

```diff
@@ -26,21 +28,22 @@ type EthereumTokenToFiatAmountFormatterProps = {
 export const EthereumTokenToFiatAmountFormatter = ({
     value,
     ethereumToken,
+    contract,
     isDiscreetText = true,
     decimals = 0,
     signValue,
     ellipsizeMode,
     numberOfLines,
     ...rest
 }: EthereumTokenToFiatAmountFormatterProps) => {
-    const coins = useSelector(selectCoinsLegacy);
     const fiatCurrency = useSelector(selectFiatCurrency);
     const { FiatAmountFormatter } = useFormatters();
+    const fiatRateKey = getFiatRateKey(ethereumToken, fiatCurrency.label, contract);
+    const rate = useSelector((state: any) => selectFiatRatesByFiatRateKey(state, fiatRateKey));
 
-    const rates = coins.find(coin => coin.symbol.toLowerCase() === ethereumToken.toLowerCase())
-        ?.current?.rates;
+    if (!rate?.rate) return <EmptyAmountText />;
```

Comment:
Should not selector return null if specific fiat rate is not found?

Then we could simplify this check:
```suggestion
    if (!rate) return <EmptyAmountText />;
```